src/main/credentialServer/httpCredentialHelperServer.ts

```diff
@@ -188,17 +261,33 @@ export class HttpCredentialHelperServer {
    * Note that callers are responsible for catching exceptions, which usually happens if the
    * `$HOME/docker/config.json` doesn't exist, its JSON is corrupt, or it doesn't have a `credsStore` field.
    */
-  protected async getCredentialHelperName(): Promise<string> {
+  protected async getCredentialHelperName(command: string, payload: string): Promise<credHelperInfo> {
     const home = findHomeDir();
     const dockerConfig = path.join(home ?? '', '.docker', 'config.json');
     const contents = JSON.parse((await fs.promises.readFile(dockerConfig, { encoding: 'utf-8' })).toString());
-    const credsStore = contents['credsStore'];
+    const credHelpers = contents.credHelpers;
+    const credsStore = contents.credsStore;
 
-    if (!credsStore) {
-      throw new Error(`No credsStore field in ${ dockerConfig }`);
+    if (credHelpers) {
+      let credsStoreOverride = '';
+
+      switch (command) {
+      case 'erase':
+      case 'get':
+        credsStoreOverride = credHelpers[payload.trim()];
+        break;
+      case 'store': {
+        const obj = JSON.parse(payload);
+
+        credsStoreOverride = obj.ServerURL ? credHelpers[obj.ServerURL] : '';
+      }
+      }
+      if (credsStoreOverride) {
+        return { credsStore: credsStoreOverride, credHelpers: { } };
+      }
     }
 
-    return credsStore;
+    return { credsStore, credHelpers };
```

Comment:
`credsStore` may be undefined here.

Reply:
The code works because when I do access the `credHelpers` field, only in `doListCommand`, it's with a `record.credHelpers ?? {}`. But less coupling if I initialize it to `{}`...